client/state/selectors/get-sites.js

```diff
@@ -20,7 +20,7 @@ export default createSelector(
 	( state ) => {
 		const primarySiteId = getPrimarySiteId( state );
 		const [ primarySite, sites ] = partition( state.sites.items, { ID: primarySiteId } );
-		return map( concat( primarySite, sites ), site => getSite( state, site.ID ) );
+		return map( concat( primarySite, sortBy( sites, [ 'name', 'URL' ] ) ), site => getSite( state, site.ID ) );
```

Comment:
The endpoint sorts the sites just by the `blogname`, so just sorting by `name` should be enough, but I think having the `URL` as a second factor is a great idea, especially for sites that don't have their name specified. :+1: